docs/user/agent/node-joiner.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+if [ $# -lt 1 ]; then
+    echo "./node-joiner.sh <pull secret path>"
+    echo "Usage example:"
+    echo "$ ./node-joiner.sh ~/config/my-pull-secret"
+
+    exit 1
+fi
+pullSecret=$1
+
+# Extract the installer image pullspec and release version.
+releaseImage=$(oc get clusterversion version -o=jsonpath='{.status.history[?(@.state == "Completed")].image}')
+nodeJoinerPullspec=$(oc adm release info -a "$pullSecret" --image-for=installer "$releaseImage")
+
+# Create the namespace to run the node-joiner, along with the required roles and bindings.
+staticResources=$(cat <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openshift-node-joiner
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: node-joiner
+  namespace: openshift-node-joiner
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: node-joiner
+rules:
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - clusterversions
+  - proxies
+  verbs:
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  - configmaps
+  - nodes
+  verbs:
+  - get
+  - list
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: node-joiner
+subjects:
+- kind: ServiceAccount
+  name: node-joiner
+  namespace: openshift-node-joiner
+roleRef:
+  kind: ClusterRole
+  name: node-joiner
+  apiGroup: rbac.authorization.k8s.io
+EOF
+)
+echo "$staticResources" | oc apply -f -
+
+# Generate a configMap to store the user configuration
+oc create configmap nodes-config --from-file=nodes-config.yaml -n openshift-node-joiner -o yaml --dry-run=client | oc apply -f -
```

Comment:
We probably want the user to pass us the filename rather than hard-coding `nodes-config.yaml`